coffeecard/CoffeeCard.WebApi/Services/TokenService.cs

```diff
@@ -35,20 +41,33 @@ public string GenerateToken(IEnumerable<Claim> claims)
 
         public JwtSecurityToken ReadToken(string token)
         {
-            return new JwtSecurityTokenHandler().ReadToken(token) as JwtSecurityToken;
+            try
+            {
+                return new JwtSecurityTokenHandler().ReadToken(token) as JwtSecurityToken;
+            }
+            catch (Exception) //In case the string token is not in a JWS or JWE Compact Serialization Format 
+            {
+                return null;
+            }
```

Comment:
Also log the exception here or somewhere else

Reply:
Would this be agreeable instead? Perhaps while logging that it received a malformed token if it can't read it?
```suggestion
            var jwtHandler = new JwtSecurityTokenHandler();
            if (jwtHandler.CanReadToken(token))
            {
                return new JwtSecurityTokenHandler().ReadToken(token) as JwtSecurityToken;
            }
            return null;
```

Reply:
My problem is more that you return null. It forces the rest of the code to do checks. I'll say just return the `return new JwtSecurityTokenHandler().ReadToken(token)`. Any exception thrown will just be propegated which would be on purpose :)

Reply:
Which forces me to handle the exception instead, or crash the website. I found it to be easier to just do a check for null, instead of having to handle the exception